### ISSUE TITLE
[Feature] deploy step supports Input/Output Variables

### ISSUE THREAD

**Issue by kferrone** [kubevela/kubevela#4658]
**Describe the bug**

If your components use input output steps, then those will not resolve if you use the deploy workflow step. 

**To Reproduce**

Try this one out:

```yaml
apiVersion: core.oam.dev/v1beta1
kind: Application
metadata:
  name: foo-tester
  namespace: default
spec:
  components: 
  - type: helm
    name: nginx-test
    inputs:
    - from: items
      parameterKey: properties.values.items
    properties:
      targetNamespace: default
      chart: nginx
      url: https://charts.bitnami.com/bitnami
      version: "12.0.0"
      values:
        replicas: 12
  - type: k8s-objects
    name: some-things
    outputs:
    - name: items
      valueFrom: output.data["items"]
    properties:
      objects:
      - apiVersion: v1
        kind: ConfigMap 
        metadata:
          name: some-things-around
        data:
          items: egss,pizza,cactus
          through: the door
          replicas: "1"
  workflow:
    steps:
    - name: upit 
      type: deploy
      properties: 
        policies: []
        
```

**Expected behavior**

The values key should ultimately have the items value. 

**Screenshots**

**KubeVela Version**

1.5

**Cluster information**

1.22 eks

**Additional context**

If you apply the components one by one with `apply-component` then the input/output is resolved.

**Comment by kferrone** [comment]
I even played around with `op.#ApplyComponent` in a worklow-step definition. The inputs and outputs nor the DAG seems to work when components are applied manually

**Comment by Somefive** [comment]
`deploy` workflowstep has some different internal implementation mechanism, therefore it does not support the input/output for components and workflow steps by far. This is a feature that we planned to do but haven't done yet.

One of the major difficulty for using input/output in `deploy` workflowstep is that it uses multi-cluster deploy, which means there are multiple inputs & outputs for a single component. This is even more difficult when we have `replication` policy used in `deploy` (in KubeVela v1.6). So in general, the `deploy` workflowstep will not implement the input/output in short term but the feature is desired exactly.

The `op.#ApplyComponent` should work fine with input/output. I'm not sure how did you use it in workflow-step definition. But it is okay to use `apply-component` workflowstep and leverage the capability of workflow input/output.

**Comment by chivalryq** [comment]
Closing with https://github.com/kubevela/kubevela/pull/5161, this feature is released in v1.7.0
